library/core/src/ptr/non_null.rs

```diff
@@ -1021,6 +1021,12 @@ impl<T: ?Sized> NonNull<T> {
     #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
     #[stable(feature = "non_null_convenience", since = "1.80.0")]
     #[rustc_const_stable(feature = "const_intrinsic_copy", since = "1.83.0")]
+    #[requires(count.checked_mul(core::mem::size_of::<T>()).map_or_else(|| false, |size| size <= isize::MAX as usize)
+        && ub_checks::can_dereference(NonNull::slice_from_raw_parts(self, count).as_ptr())
+        && ub_checks::can_write(NonNull::slice_from_raw_parts(dest, count).as_ptr()))]
+    #[ensures(|result: &()| ub_checks::can_dereference(self.as_ptr() as *const u8)
+        && ub_checks::can_dereference(dest.as_ptr() as *const u8))]
+    #[kani::modifies(NonNull::slice_from_raw_parts(dest, count).as_ptr())]
```

Comment:
This now fails to compile.

Reply:
I just updated the branch with main and does successfully run, could you please check again?
````
Checking harness ptr::non_null::verify::non_null_check_copy_to_nonoverlapping...

VERIFICATION RESULT:
 ** 0 of 377 failed (2 unreachable)

VERIFICATION:- SUCCESSFUL
Verification Time: 6.860433s

Checking harness ptr::non_null::verify::non_null_check_copy_to...

VERIFICATION RESULT:
 ** 0 of 340 failed (4 unreachable)

VERIFICATION:- SUCCESSFUL
Verification Time: 6.714273s

Complete - 2 successfully verified harnesses, 0 failures, 2 total.`
```

Reply:
This still fails, see https://github.com/model-checking/verify-rust-std/actions/runs/12267053169/job/34226426202?pr=149.